### ISSUE TITLE
Issue #6

### ISSUE THREAD

**Issue by zane-shus** [mwgg/iracing-php#7]
hay mate, just checking you saw the updates to Issue #6    Not sure it alerts commenting on a closed ticket, so raising a new so you get an alert.

**Comment by kylecrouse** [comment]
The `body->data` vs `body->chunk_info` change in `retrieveChunks()` is one issue, but that chunk resolver function is never getting called in the current implementation. 

Updating the `request()` method to resolve chunks, if needed, after the link data has been retrieved similar to this snippet will return the data. 

This may not be a scalable solution when considering the chunk data will likely be quite large when dealing with something like a Nurburgring endurance race.

```    
public function request(string $endpoint, array $data = [])
    {
        $url = LibConstants::APIURL . $endpoint;
        try
        {
            $response = $this->guzzle->request('GET', $url, [
                'cookies' => $this->jar,
                'query' => $data
            ]);

            $this->setRateLimits($response);

            $responseBody = json_decode($response->getBody());
            if(isset($responseBody->link))
            {
                $responseData = $this->retrieveData($responseBody->link);
                if(isset($responseData->chunk_info) && $responseData->chunk_info->num_chunks > 0)
                {
                    $responseData->chunks = $this->retrieveChunks($responseData);
                }
                return $responseData;
            }
        }
        catch(\GuzzleHttp\Exception\BadResponseException $e)
        {
            $response = $e->getResponse();
            $this->setRateLimits($response);
            if($this->shouldRetryRequest($response, $e->getMessage(), $e))
            {
                return $this->request($endpoint, $data);
            }
        }
        return null;
    }
```

**Comment by mwgg** [comment]
Hi, sorry for not getting around to this sooner.

The issue was that not all endpoints behave in the same way with chunks. For example, `search_series` returns the chunks data directly and that worked fine, but `event_log` returns an AWS link like many other endpoints, and that contains some regular data *plus* the chunks. So the chunks worked, but not on all endpoints.

I have fixed this with v1.6.0, any nested chunks are now retrieved and placed into the `data` attribute of the response.
